lib/server/swarm.js

```diff
@@ -29,15 +29,7 @@ function Swarm (infoHash, server) {
       peer_id: peer.peerId
     }
     self._onAnnounceStopped(params, peer, peer.peerId)
-
-    // When a websocket peer is evicted, and it's not in any other swarms, close
-    // the websocket to conserve server resources.
-    if (peer.socket && peer.socket.infoHashes.length === 0) {
-      try {
-        peer.socket.close()
-        peer.socket = null
-      } catch (err) {}
-    }
+    peer.socket = null
```

Comment:
In this situation, either way works fine. The whole `peer` object is going to be garbage-collected once all the references to it are gone.

The situation you're referring to is when trying to remove a property from an instance of a class (i.e. something with a prototype). In that case, it's better to use `object.prop = null` since `delete object.prop` will cause a de-optimization because it changes the "shape" of the object in memory.

Using `delete` is fine on object literals `{}` though.